### PR TITLE
Update SecurityController to not use deprecated classes

### PR DESCRIPTION
**Pull request by xelaris**
This PR avoids usage of deprecated `SecurityContextInterface` and `form.csrf_provider` service in the `SecurityController` to enable usage of the controller in SF 2.7 without triggering deprecation warnings.

### CODE REVIEW

**Comment by Soullivaneuh** [comment]
:+1:

File: Controller/SecurityController.php [hunk 1]
```diff
@@ -38,15 +48,20 @@ public function loginAction(Request $request)
...
 
         // last username entered by the user
-        $lastUsername = (null === $session) ? '' : $session->get(SecurityContextInterface::LAST_USERNAME);
+        $lastUsername = (null === $session) ? '' : $session->get($lastUsernameKey);
 
-        $csrfToken = $this->has('form.csrf_provider')
-            ? $this->get('form.csrf_provider')->generateCsrfToken('authenticate')
-            : null;
+        if ($this->has('security.csrf.token_manager')) {
+            $csrfToken = $this->get('security.csrf.token_manager')->getToken('authenticate');
```
Comment (stof) [inline Controller/SecurityController.php]: should be ``->getToken('authenticate')->getValue()``. ``getToken`` returns a CsrfToken object, not a string

**Comment by xelaris** [comment]
Thanks @stof. I added the `getValue()` call. I didn't notice it, as the `CsrfToken` provides a `__toString()` method, so passing a `CsrfToken` object to the template works as expected.

**Comment by stof** [comment]
Thank you @xelaris.

**Pull request merged**